statistics/handle/cache/internal/lfu/lfu_cache_test.go

```diff
@@ -170,6 +171,71 @@ func TestLFUCachePutGetWithManyConcurrency2(t *testing.T) {
 	require.Equal(t, 1000, len(lfu.Values()))
 }
 
+func TestLFUCachePutGetWithManyConcurrencyAndSmallConcurrency(t *testing.T) {
```

Comment:
Can this test reproduce DATA RACE on the Master branch?

Reply:
I put this test into master and found it.

```
==================
WARNING: DATA RACE
Write at 0x00c00779d250 by goroutine 26:
  github.com/pingcap/tidb/statistics.(*Index).DropUnnecessaryData()
      /Users/weizhenwang/devel/opensource/tidb/statistics/index.go:74 +0x78
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.DropEvicted()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache.go:123 +0x70
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.(*LFU).onEvict()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache.go:146 +0x1a4
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.(*LFU).onReject()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache.go:128 +0x58
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.(*LFU).onReject-fm()
      <autogenerated>:1 +0x3c
  github.com/dgraph-io/ristretto.NewCache.func3()
      /Users/weizhenwang/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:194 +0x70
  github.com/dgraph-io/ristretto.(*Cache).processItems()
      /Users/weizhenwang/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:475 +0x598
  github.com/dgraph-io/ristretto.NewCache.func5()
      /Users/weizhenwang/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x34

Previous read at 0x00c00779d250 by goroutine 33:
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.checkTable()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache_test.go:261 +0x2e0
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.TestLFUCachePutGetWithManyConcurrencyAndSmallConcurrency.func2()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache_test.go:229 +0xf4

Goroutine 26 (running) created at:
  github.com/dgraph-io/ristretto.NewCache()
      /Users/weizhenwang/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0xc28
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.NewLFU()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache.go:54 +0x3e0
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.TestLFUCachePutGetWithManyConcurrencyAndSmallConcurrency()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache_test.go:206 +0x30
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1595 +0x194
  testing.(*T).Run.func1()
      /opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1648 +0x40

Goroutine 33 (running) created at:
  github.com/pingcap/tidb/statistics/handle/cache/internal/lfu.TestLFUCachePutGetWithManyConcurrencyAndSmallConcurrency()
      /Users/weizhenwang/devel/opensource/tidb/statistics/handle/cache/internal/lfu/lfu_cache_test.go:223 +0x160
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1595 +0x194
  testing.(*T).Run.func1()
      /opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1648 +0x40
==================

```

Reply:
Cool